app/models/hubstats/team.rb

```diff
@@ -0,0 +1,150 @@
+      select("hubstats_teams.id as team_id")
+      .select("IFNULL(COUNT(DISTINCT hubstats_pull_requests.id),0) AS pull_request_count")
+      .joins(sanitize_sql_array(["LEFT JOIN hubstats_pull_requests ON hubstats_pull_requests.team_id = hubstats_teams.id AND (hubstats_pull_requests.merged_at BETWEEN ? AND ?) AND hubstats_pull_requests.merged = '1'", start_date, end_date]))
+      .group("hubstats_teams.id")
+    }
+
+    # Public - Counts all of the addtiions and deletions made from PRs by a member of the selected team that have been merged between the start_date
+    # and the end_date.
+    #
+    # start_date - the start of the date range
+    # end_date - the end of the data range
+    #
+    # Returns - the additions and deletions
+    scope :net_additions_count, lambda {|start_date, end_date|
+      select("hubstats_teams.id as team_id")
+      .select("SUM(IFNULL(hubstats_pull_requests.additions, 0)) AS additions")
+      .select("SUM(IFNULL(hubstats_pull_requests.deletions, 0)) AS deletions")
+      .joins(sanitize_sql_array(["LEFT JOIN hubstats_pull_requests ON hubstats_pull_requests.team_id = hubstats_teams.id AND (hubstats_pull_requests.merged_at BETWEEN ? AND ?) AND hubstats_pull_requests.merged = '1'", start_date, end_date]))
+      .group("hubstats_teams.id")
+    }
+
+    # Public - Joins all of the metrics together for selected team: net additions, comments, repos, pull requests, and users.
+    # 
+    # start_date - the start of the date range
+    # end_date - the end of the data range
+    # 
+    # Returns - all of the stats about the team
+    scope :with_all_metrics, lambda {|start_date, end_date|
+      select("hubstats_teams.*, user_count, pull_request_count, comment_count, repo_count,additions, deletions")
+       .joins("LEFT JOIN (#{net_additions_count(start_date, end_date).to_sql}) AS net_additions ON net_additions.team_id = hubstats_teams.id")
+       .joins("LEFT JOIN (#{pull_requests_count(start_date, end_date).to_sql}) AS pull_requests ON pull_requests.team_id = hubstats_teams.id")
+       .joins("LEFT JOIN (#{comments_count(start_date, end_date).to_sql}) AS comments ON comments.team_id = hubstats_teams.id")
+       .joins("LEFT JOIN (#{repos_count(start_date, end_date).to_sql}) AS repos ON repos.team_id = hubstats_teams.id")
+       .joins("LEFT JOIN (#{users_count(start_date, end_date).to_sql}) AS users ON users.team_id = hubstats_teams.id")
+       .group("hubstats_teams.id")
+    }
+
+    attr_accessible :name, :hubstats
+    has_and_belongs_to_many :users, :join_table => 'hubstats_teams_users', :uniq => true
+    
+    # Public - Checks if the team is currently existing, and if it isn't, then makes a new team with 
+    # the specifications that are passed in. We are assuming that if it is not already existent,
+    # then we probably don't really care about the team, so our hubstats boolean will be set to false.
+    #
+    # github_team - the info that's passed in about the new team
```

Comment:
github_team is a membership event payload right, not a new team event?
https://developer.github.com/v3/activity/events/types/#memberevent

Consider updating for clarity.